src/main/java/org/testobject/appium/testng/TestObjectTestNGTestProvider.java

```diff
@@ -0,0 +1,55 @@
+package org.testobject.appium.testng;
+
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.URL;
+
+public class TestObjectTestNGTestProvider {
+
+	private RemoteWebDriver remoteWebDriver;
+
+	private URL endpoint;
+
+	private boolean isLocalTest;
+
+	public static TestObjectTestNGTestProvider newInstance() {
+		return new TestObjectTestNGTestProvider();
+	}
+
+	private TestObjectTestNGTestProvider() {
+	}
+
+	public RemoteWebDriver getRemoteWebDriver() {
+		return remoteWebDriver;
+	}
+
+	public AppiumDriver getAppiumDriver() {
+		if (remoteWebDriver != null) {
```

Comment:
if you want to do nicer if else check following recommendations:
favorite: return (a!=null?(AppiumDriver)a:null)
second best:
if(a==null){
  return null;
}
return (AppiumDriver) a;
--> try to not indent the positive control flow but the error case to prevent unnecessary indention in the whole control flow

(Just wanna share my opinion with you on how clean code could look like)